src/core/mesh/qgsmeshlayerutils.h

```diff
@@ -251,17 +251,14 @@ class CORE_EXPORT QgsMeshLayerUtils
     static QgsRectangle triangleBoundingBox( const QgsPointXY &p1, const QgsPointXY &p2, const QgsPointXY &p3 );
 
     /**
-     * Formats hours in human readable string based on settings
+     * Formats hours in human readable string based on settings and reference time
+     * If reference time is invalid, return relative time
+     * \param hours time in hours from reference time
+     * \param referenceTime the reference time
+     * \param settings the time settings
+     * \return the formatted time
      */
-    static QString formatTime( double hours, const QgsMeshTimeSettings &settings );
-
-    /**
-      * Searches and returns the first valid reference time in layer's dataset group
-      * \param meshLayer mesh layer to parse
-      *
-      * \since QGIS 3.12
-      */
-    static QDateTime firstReferenceTime( QgsMeshLayer *meshLayer );
+    static QString formatTime( double hours, const QDateTime &referenceTime, const QgsMeshTimeSettings &settings );
```

Comment:
Do we need this QgsMeshTimeSettings at all? (these are only for static datasets, right?)...

if this function is only for static dataset and that temporal widgets where it is set, I would remove it from the qgsmeshlayerutils.h and put it in the qgis_app so it is not in the public API

Reply:
It seems to be used too with mesh calculator for the time format... Needs some refactoring and I wonder if we could not put the time format setting in the QgsSettings?

Reply:
Best to ask Nyall in the official PR